scmail.py

```diff
@@ -134,42 +143,61 @@ def register(ctx, fingerprint):
 
 
 @click.command()
-@click.option('--fingerprint', '-f', prompt='Enter fingerprint of mailbox', required=True, help='The fingerprint of the yourself mailbox.')
-@click.option('--sender-fingerprint', '-s', prompt='Enter the fingerprint of sender', default=None, help='The senders fingerprint.')
-@click.password_option('--password', '-p', prompt='Enter password of private key', help='The passphrase of private key')
+@click.option('--fingerprint', '-f', prompt='Enter fingerprint of mailbox',
+              required=True, type=str, help='The fingerprint of the yourself mailbox.')
+@click.option('--sender-fingerprint', '-s', prompt='Enter the fingerprint of sender',
+              type=str, help='The senders fingerprint.')
+@click.password_option('--password', '-p', prompt='Enter password of private key',
+                       help='The passphrase of private key')
 @click.pass_context
 def retrieve(ctx, fingerprint, sender_fingerprint, password):
     """Retrieve and Post messages from API"""
+    # Prepare send data.
     logging.debug(fingerprint, sender_fingerprint)
     data = {'fingerprint': fingerprint}
-    if not sender_fingerprint:
+    if sender_fingerprint == '':
         data.update(sender_fingerprint, sender_fingerprint)
-    # Retrieve
+
+    # Retrieve from api.
     r = requests.post(SECUREMAILBOX_URL + '/retrieve/', json=data)
     res = r.json()
-    logging.debug(f'response is: {res}')
 
     if r.status_code == 200:
         logging.info('The message retrieve successful.')
     else:
         logging.error(f'The message retrieve fail.\nError {r.status_code} is: {res.get("error")}')
         return
 
+    # load messages.
+    logging.debug(f'response is: {res}')
+    all_messages = res.get('data').get('messages')
+    logging.debug([message.get('message') for message in all_messages])
+
     # Decrypt the messages.
     ok, messages = ctx.parent.gpg.decrypt_message(messages=res.get('error'), passphrase=password)
-    ctx.parent.pp.pprint(messages[:-1])
+    logging.debug(messages)
+
+    # Print all message.
+    for m in messages:
+        all_messages[messages.index(m)]['message'] = m
```

Comment:
I'm confused by this. It looks like this grabs all of the messages from the request and then asks GPG to decrypt the `error` field of the request. It then overwrites all messages in the response object's messages field. It's probably better practice (and a little more efficient) to just print the decrypted messages returned from GPG.

Reply:
Sorry about that, I think that is the previous code (before retrieve is available). The logic (in this version) is the GPG decrypt all messages and print as original API format.
Original API format:
```
created_at: "Mon, 13 Apr 2020 22:41:46 GMT"
message: "This is a messge"
sender_fingerprint": "8908218A320019CC0F37C80630C9054FDFA961B1"
updated_at: "Mon, 13 Apr 2020 22:41:46 GMT"

created_at: "Mon, 13 Apr 2020 22:41:46 GMT"
message: "Also a message."
sender_fingerprint": "8908218A320019CC0F37C80630C9054FDFA961B1"
updated_at: "Mon, 13 Apr 2020 22:41:46 GMT"
```
If just print message:
```
This is a message.

Also a message.
```
But this work for me. I changed to second format.